docs/USAGE-GUIDE.md

```diff
@@ -77,81 +77,92 @@ And when invoking CMake for your project, do one of two things:
-The previous instructions work fine with only a few tweaks to your config command. Example:
+
+*OpenSSL-1.0.2*
 ```shell
-setarch i386 ./config -fPIC no-shared     \
-        -m32 no-md2 no-rc5 no-rfc3779 no-sctp no-ssl-trace no-zlib     \
-        no-hw no-mdc2 no-seed no-idea no-camellia\
-        no-bf no-ripemd no-dsa no-ssl2 no-ssl3 no-capieng     \
-        -DSSL_FORBID_ENULL -DOPENSSL_NO_DTLS1 -DOPENSSL_NO_HEARTBEATS   \
-        --prefix=`pwd`/../../libcrypto-root/
+curl -LO https://github.com/openssl/openssl/archive/refs/tags/OpenSSL_1_0_2.tar.gz
+tar -xzvf OpenSSL_1_0_2.tar.gz
+cd `tar ztf OpenSSL_1_0_2.tar.gz | head -n1 | cut -f1 -d/`
 ```
 
-## Building s2n-tls with OpenSSL-1.0.2
+### Build Openssl
+The following config command disables numerous Openssl features and algorithms which are not used
+by s2n-tls. A minimal feature-set can help prevent exposure to security vulnerabilities.
+
+*OpenSSL-1.1.1* and *OpenSSL-3.0.5*
+```shell
+./config -fPIC no-shared              \
+        no-md2 no-rc5 no-rfc3779 no-sctp no-ssl-trace no-zlib     \
+        no-hw no-mdc2 no-seed no-idea enable-ec_nistp_64_gcc_128 no-camellia\
+        no-bf no-ripemd no-dsa no-ssl2 no-ssl3 no-capieng                  \
+        -DSSL_FORBID_ENULL -DOPENSSL_NO_DTLS1 -DOPENSSL_NO_HEARTBEATS      \
+        --prefix=`pwd`/../../libcrypto-root/
 
-To build s2n-tls with OpenSSL-1.0.2, do the following:
+make
+make install
+```
 
+*OpenSSL-1.0.2*
 ```shell
-# We keep the build artifacts in the -build directory
-cd libcrypto-build
+./config -fPIC no-shared              \
+        no-libunbound no-gmp no-jpake no-krb5 no-store    \
+        no-md2 no-rc5 no-rfc3779 no-sctp no-ssl-trace no-zlib     \
+        no-hw no-mdc2 no-seed no-idea enable-ec_nistp_64_gcc_128 no-camellia\
+        no-bf no-ripemd no-dsa no-ssl2 no-ssl3 no-capieng                  \
+        -DSSL_FORBID_ENULL -DOPENSSL_NO_DTLS1 -DOPENSSL_NO_HEARTBEATS      \
+        --prefix=`pwd`/../../libcrypto-root/
 
-# Download the latest version of OpenSSL
-curl -LO https://www.openssl.org/source/openssl-1.0.2-latest.tar.gz
-tar -xzvf openssl-1.0.2-latest.tar.gz
-
-# Build openssl libcrypto
-cd `tar ztf openssl-1.0.2-latest.tar.gz | head -n1 | cut -f1 -d/`
-./config -fPIC no-shared no-libunbound no-gmp no-jpake no-krb5              \
-         no-md2 no-rc5 no-rfc3779 no-sctp no-ssl-trace no-store no-zlib     \
-         no-hw no-mdc2 no-seed no-idea enable-ec-nistp_64_gcc_128 no-camellia\
-         no-bf no-ripemd no-dsa no-ssl2 no-ssl3 no-capieng                  \
-         -DSSL_FORBID_ENULL -DOPENSSL_NO_DTLS1 -DOPENSSL_NO_HEARTBEATS      \
-         --prefix=`pwd`/../../libcrypto-root/
 make depend
 make
 make install
+```
 
-# Build s2n-tls
-cd ../../
+### Build s2n-tls
+```shell
+cd ../../ # root of project
 make
 ```
 
-**Mac Users:** please replace "./config" with "./Configure darwin64-x86_64-cc".
+### Alternative platforms
+*OpenSSL-1.1.1 32-bit builds* should use the following config command:
```

Comment:
I'd put this as either a subsection of "### Build Openssl", or just other options under  "### Build Openssl" like the existing two.

Reply:
fair point